### PR TITLE
feat: add reserved column

### PR DESCRIPTION
**Pull request by jiacai2050**
## Rationale
In order to maintain compatibility, add a reserved column for future use.

## Test Plan
CI

### CODE REVIEW

File: src/metric_engine/src/types.rs [hunk 1]
```diff
@@ -132,18 +140,107 @@ pub struct WriteResult {
...
+    }
+
+    pub fn is_builtin_field(f: &FieldRef) -> bool {
+        f.name() == SEQ_COLUMN_NAME || f.name() == RESERVED_COLUMN_NAME
+    }
+
+    pub fn fill_builtin_projections(&self, projection: &mut Option<Vec<usize>>) {
+        if let Some(proj) = projection.as_mut() {
+            for i in 0..self.num_primary_keys {
+                if !proj.contains(&i) {
```
Comment (zealchen) [inline src/metric_engine/src/types.rs:205]: primary keys are not builtin columns, why put them in here?

Comment (jiacai2050) [reply src/metric_engine/src/types.rs:205]: Primary keys column are required when query, I will rename this function to `fill_required_projections`

**Comment by zealchen** [comment]
LGTM

**Review by zealchen — APPROVED** [review]
LGTM

**Pull request merged**